source/material_model/rheology/strain_dependent.cc

```diff
@@ -502,6 +502,7 @@ namespace aspect
         // Calculate changes in strain and update the reaction terms
         if  (this->simulator_is_past_initialization() && this->get_timestep_number() > 0 && in.requests_property(MaterialProperties::reaction_terms))
           {
+            Assert(in.strain_rate.size()>0, ExcInternalError());
```

Comment:
Is this assert just to provide an additional error message in the event that the conditions are met on line 503 but the strain rate is still empty? If so, would it be worth adding a more detailed error message or a comment above?

Reply:
This assert is the only reliable way to trigger an error when you illegally access in.strain_rate[0] when it is empty. We should probably put it in a few more places inside material models just in case.
I can make the message better of course.

Reply:
Do you think this is the only place in ASPECT where we use the strain rate for something unrelated to the viscosity?